### DISCUSSION TITLE
Outdated Markdown package in Overleaf?

### DISCUSSION THREAD

**Discussion by michaelstepner** [Witiko/markdown#251] [Q&A] [answered]
Hello! I've been trying (and failing) to use a custom lua function to pre-process text before it is processed by the Markdown package. That process of experimentation has led me to a more specific question, so I'll make a separate post for each topic.

I concluded I should try using the `extensions()` option [described in the User Manual](https://muug.ca/mirror/ctan/macros/generic/markdown/markdown.html#option-extensions), and have tried to implement the minimal LaTeX example given in the docs. When I try to run it in Overleaf, I get the following error and the extension does not work:

![image](https://user-images.githubusercontent.com/6315497/213605861-74361dc7-b907-4c95-83c1-da5e3266f5dd.png)

![image](https://user-images.githubusercontent.com/6315497/213605922-6087a7b5-7515-47be-8694-6fcddff0a02a.png)

I suspect that might be because the version of the Markdown package loaded by Overleaf from TeX Live 2022 is outdated, and might predate the implementation of the `extensions()` option:

```
Package: markdown 2022/05/31 v2.15.2-0-gb238dbc markdown renderer
```

Is that accurate? If so, do you have advice on how to use the latest version of the package from within Overleaf?

**Comment by Witiko** [comment, marked as answer]
Hello, @michaelstepner. As you discovered, Overleaf tends to use outdated TeX Live images. To use the current version of the Markdown package on Overleaf, you can add a copy of the package to your Overleaf project. Sections [Installation][1] and [Portable installation][2] of the user manual outline the necessary steps.

 [1]: https://witiko.github.io/markdown/#installation
 [2]: https://witiko.github.io/markdown/#portable-installation

**Reply by michaelstepner** [reply]
@Witiko thanks, I was able to successfully upgrade to the current version of the Markdown package on Overleaf by following your instructions and [updating my `latexmkrc` file](https://www.overleaf.com/learn/latex/Questions/I_have_a_lot_of_.cls%2C_.sty%2C_.bst_files%2C_and_I_want_to_put_them_in_a_folder_to_keep_my_project_uncluttered._But_my_project_is_not_finding_them_to_compile_correctly).

Here is the resulting Overleaf project: https://www.overleaf.com/read/pwsbymsxhnvw

And I can see in the logs that it's now using the current version of the Markdown package:

```
Package: markdown 2023-01-14 v2.19.0-58-g91b8caf markdown renderer
```

Unfortunately, when I typeset the minimal example of the `extensions()` option [described in the User Manual](https://muug.ca/mirror/ctan/macros/generic/markdown/markdown.html#option-extensions), it's still not successfully striking through the text:

<img width="1093" alt="image" src="https://user-images.githubusercontent.com/6315497/213775403-dec0811e-56cc-4fef-9575-ac2c62c6bd79.png">

Do you have any suggestions for why the example isn't working? If you'd like me to open a separate discussion since we've resolved the "Outdated Markdown package" issue, let me know.

**Reply by Witiko** [reply]
> and [updating my latexmkrc file](https://www.overleaf.com/learn/latex/Questions/I_have_a_lot_of_.cls%2C_.sty%2C_.bst_files%2C_and_I_want_to_put_them_in_a_folder_to_keep_my_project_uncluttered._But_my_project_is_not_finding_them_to_compile_correctly).

I think that's your issue. TeX will find the `.tex` and `.sty` files in the `markdown/` directory as per the `TEXINPUTS` variable but not the Lua files. You may need to place the files `markdown/markdown.lua` and `markdown/libraries/markdown-tinyyaml.lua` into your root directory. In general, you should follow [the portable installation instructions][1] in the user manual.

 [1]: https://witiko.github.io/markdown#installation
